packages/program/src/functions/base-invocation-scope.ts

```diff
@@ -292,9 +293,10 @@ export class BaseInvocationScope<TReturn = any> {
     assert(this.program.account, 'Wallet is required!');
 
     const transactionRequest = await this.getTransactionRequest();
-
     const { maxFee } = await this.getTransactionCost();
 
+    await this.setDefaultTxParams();
+
     await this.fundWithRequiredCoins(maxFee);
 
```

Comment:
```suggestion
    await this.setDefaultTxParams();

    const { maxFee } = await this.getTransactionCost();
    await this.fundWithRequiredCoins(maxFee);

    const transactionRequest = await this.getTransactionRequest();
```

At first read I had trouble understanding the flow of logic because `transactionRequest` was initialized and only used later down the line while some additional transaction modifications were happening with calls to `this.setDefaultTxParams` and `this.fundWithRequiredCoins`. The catch is that `transactionRequest` is a pointer to `this.transactionRequest`. I think that the suggesed order of logic is more natural: modify state of transaction, get `transactionRequest`, send it.